notion-core/src/main/scala/com/ambiata/notion/core/PartReducer.scala

```diff
@@ -0,0 +1,69 @@
+package com.ambiata.notion.core
+
+import java.security.MessageDigest
+
+import com.ambiata.mundane.io.Checksum
+import org.apache.commons.codec.binary.Base64
+
+import scala.collection.mutable.Queue
+import scalaz.{Foldable, Monoid}
+
+/**
+ * "Reducer-like" class to accumulate values while
+ * reading "parts" (bytes, lines) from a file and transforming them to a final value
+ * when the whole file has been read
+ */
+trait PartReducer[P, T] {
+  type S
+  
+  /** initial value */
+  def init: S
+
+  /** reducer function using the current part and the current state S */
+  def reduce(part: P, s: S): S
+
+  /** final value from the last state */
+  def finalise(s: S): T
+}
```

Comment:
Happy for this to be postponed to a separate discussion / PR, but I think this should just be called Reducer or Fold, and should be in a library of its own, or in a mundane-fold module until it graduates. It should also look at some of the other similar implementation to take into account the naming / specialization that is going in. But lets get this in first.